File: 657-robot-return-to-origin/robot-return-to-origin.rs
impl Solution {
    pub fn judge_circle(moves: String) -> bool {
        let mut x = 0;
        let mut y = 0;

        for move_char in moves.chars() {
            match move_char {
                'R' => x += 1,
                'L' => x -= 1,
                'U' => y += 1,
                'D' => y -= 1,
                _ => {}
            }
        }

        x == 0 && y == 0
    }
}

File: 657-robot-return-to-origin/robot-return-to-origin.cpp
class Solution {
 public:
  bool judgeCircle(string moves) {
    int x = 0, y = 0;

    for (char& move : moves) {
      if (move == 'R') x++;
      if (move == 'L') x--;
      if (move == 'U') y++;
      if (move == 'D') y--;
    }

    return x == 0 && y == 0;
  }
};